packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -453,17 +459,25 @@ static String interpretMarkerOptions(Object o, MarkerOptionsSink sink, CozyMarke
     }
   }
 
-  private static void interpretCozyMarkerData(MarkerOptionsSink sink, Map<String, Object> cozyMarkerData, CozyMarkerBuilder cozyMarkerBuilder) {
-    final Bitmap bitmap = cozyMarkerBuilder.buildMarker(new CozyMarkerData(
-      (String) cozyMarkerData.get("label"),
-      (String) cozyMarkerData.get("icon"),
-      (boolean) cozyMarkerData.get("hasPointer"),
-      (boolean) cozyMarkerData.get("isSelected"),
-      (boolean) cozyMarkerData.get("isVisualized"),
-      (String) cozyMarkerData.get("state"),
-      (String) cozyMarkerData.get("variant"),
-      (String) cozyMarkerData.get("size")
-    ));
+  /** Returns the dartMarkerId of the interpreted marker. */
+  static String interpretMarkerOptions(Object o, MarkerOptionsSink sink, CozyMarkerBuilder cozyMarkerBuilder) {
+    final String markerId = interpretMarkerOptionsWithoutIcon(o, sink);
+    final Map<?, ?> data = toMap(o);
```

Comment:
To change this, I would need to change the function as well (otherwise it would be just a cast), and it would change in a lot other places. Seeing the commit log of plugins repo, it has been like that for a while (maybe for some reason), so I will not bother to change this and possibly break something